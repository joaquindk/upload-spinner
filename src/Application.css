.container {
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}

.demoWrapper {
  display: grid;
  grid-template-rows: 200px auto;
  grid-row-gap: 32px;
}

.controls {
  display: grid;
  grid-auto-flow: column;
  grid-column-gap: 16px;
}

.controls button {
  width: 150px
}
