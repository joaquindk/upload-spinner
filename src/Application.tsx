import React, { useState, useLayoutEffect } from 'react'
import ProgressSpinner from './components/progressSpinner'
import styles from './Application.css'

const MAX_PROGRESS = 3000

/** Demo application with the required logic to demo the progress spinner. */
const Application = () => {
  const [isSpinning, setIsSpinning] = useState(true)
  const [currentProgress, setCurrentProgress] = useState(0)
  const [isSimulating, setIsSimulating] = useState(false)

  useLayoutEffect(() => {
    if (isSimulating && currentProgress < MAX_PROGRESS) {
      setTimeout(() => {
        setCurrentProgress(currentProgress + 1)
      })
    } else if (!isSimulating) {
      setCurrentProgress(0)
    }
  }, [isSimulating, currentProgress])

  return (
    <div className={styles.container}>
      <div className={styles.demoWrapper}>
        <ProgressSpinner progress={currentProgress} total={MAX_PROGRESS} spin={isSpinning} />
        <div className={styles.controls}>
          <button onClick={() => setIsSimulating(!isSimulating)}>
            {isSimulating ? 'Reset simulation' : 'Start simulation'}
          </button>
          <button onClick={() => setIsSpinning(!isSpinning)}>
            {isSpinning ? 'Stop spinning' : 'Start spinning'}
          </button>
        </div>
      </div>
    </div>
  )
}

export default Application
