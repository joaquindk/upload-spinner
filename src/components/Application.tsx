import React from 'react'

const Application: React.FunctionComponent<{}> = () => (
  <h1>Write your app here</h1>
)

export default Application
