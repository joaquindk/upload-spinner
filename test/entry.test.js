import React from 'react'
import Application from 'src/components/Application'
import { render } from '@testing-library/react'

describe('Application', () => {
  it('should render the application', () => {
    const { container } = render(<Application />)

    expect(container).toMatchSnapshot()
  })
})
