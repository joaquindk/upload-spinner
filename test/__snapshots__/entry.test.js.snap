// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application should render the application 1`] = `
<div>
  <h1>
    Write your app here
  </h1>
</div>
`;
